src/Data/Maybe/Properties.agda

```diff
@@ -97,6 +97,10 @@ maybe′-map : ∀ j (n : C) (f : A → B) ma →
              maybe′ j n (map f ma) ≡ maybe′ (j ∘′ f) n ma
 maybe′-map = maybe-map
 
+maybe-∘ : ∀ {c x} (f : B → C) (g : A → B) → f (maybe g c x) ≡ maybe (f ∘ g) (f c) x
+maybe-∘ {x = just _}  _ _ = refl
```

Comment:
Pattern-matching on an implicit argument like this doesn't quite smell right... and in any case, this lemma can probably (better) be given a direct one-line definition, again in terms of `maybe`...?

Reply:
Also, on the model of `maybe′-map`, should the lemma be stated (and named) explicitly involving `maybe′` rather than `maybe`?

Reply:
Here's a proof that uses `maybe` instead of matching. I think it's clearly worse, and I don't think it can be reduced further:
```
maybe-∘ {x = x} f _ = maybe {B = λ y → f (maybe _ _ y) ≡ maybe _ _ y} (λ _ → refl) refl x
```
I don't mind converting `x` to an explicit argument, but there's plenty of good reasons to match on a hidden one, so I don't see that as a reason to do so.

Reply:
Hmm, I'm not sure it's worth weakening the resulting statement just so that the proof is a bit simpler. And arguably it's not even actually simpler (up to inlining), it's just a difference in the surface language code.

Reply:
Not sure I understand how this is 'weaker' as a statement? In any downstream client use where `f` and/or `g` might depend on a given `x`, then appropriate arguments to the lemma can still be supplied, or not? Their *types* don't depend on `x`, so each of the arguments is 'independent' in the telescope?

Reply:
Ah, yes, I was confused. We're not actually using the more general type of `maybe` here.